src/x264_publisher.cpp

```diff
@@ -192,37 +192,44 @@ namespace x264_image_transport {
         if (!encFrame_)
         {
             ROS_ERROR("Cannot allocate frame");
-            //Cleanup memory            
+            //Cleanup memory
             memory_cleanup();
             pthread_mutex_unlock (&mutex_);
             return;
         }
 
+        encFrame_->width = width;
+        encFrame_->height = height;
+
         // Prepare the software scale context
         // Will convert from RGB24 to YUV420P
         if (encoding == enc::BGR8)
```

Comment:
Not for this PR, but it seems that this could be simplified a lot doing this:
``` c++
if (encoding == enc::BGR8)
{
  encFrame_->format = AV_PIX_FMT_BGR24;
} else if (encoding == enc::RGB8)
{
  encFrame_->format = AV_PIX_FMT_RGB24;
}  ...
```

for all "good" branches, and then:

``` c++
            sws_ctx_ = sws_getContext(width, height, encFrame_->format, //src
                                      encCdcCtx_->width, encCdcCtx_->height, encCdcCtx_->pix_fmt, //dest
                                      SWS_FAST_BILINEAR, NULL, NULL, NULL);
```

Of even creating an std::map with all the mappings, so instead of the if-else-if blocks, you can just do something like this:
``` c++
const auto format = encodingFormatsMap.find(encoding);

if (format == encodingFormatsMaps.end())
{
  if (encoding == enc::BAYER_GBRG16)
  {
    // THE CODE FOR THIS NOT SUPPORTED YET ENCODING
  }
  else
  {
    // THE CODE FOR UNKNOWN NON-SUPPORTED ENCODING
  }
}
```